exercise-book/src/nrf52-radio-panicking.md

```diff
@@ -46,7 +46,7 @@ stack backtrace:
 
 In `no_std` programs the behavior of panic is defined using the `#[panic_handler]` attribute. In the example, the *panic handler* is defined in the `panic_log` crate but we can also implement it manually:
 
-✅ Comment out the `use radio as _;` import and add the following function to the example:
+✅ Comment out the `use radio_app as _;` import and add the following function to the example:
```

Comment:
On Windows, I eliminated the `use radio_app as _;` import and copy-pasted the `panic_handler` given below.

Am I supposed to be getting this error after doing that, or do I need to also fiddle with the `flip-link`?

```
PS C:\Users\mrg\work\rust-exercises\nrf52-code\radio-app> cargo run --bin panic
   Compiling radio_app v0.0.0 (C:\Users\mrg\work\rust-exercises\nrf52-code\radio-app)
error: linking with `flip-link` failed: exit code: 0xc0000374
  |
  = note: "flip-link" "-flavor" "gnu" "C:\\Users\\mrg\\AppData\\Local\\Temp\\rustc6JKiAt\\symbols.o" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\deps\\panic-9169f66aa766dd2b.panic.52730e7b31a8b40d-cgu.0.rcgu.o" "--as-needed" "-L" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\deps" "-L" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\debug\\deps" "-L" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\build\\cortex-m-fc61aa674606e95d\\out" "-L" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\build\\cortex-m-rt-7a75c9816154ad6c\\out" "-L" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\build\\defmt-4dc4d6117a44c56c\\out" "-L" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\build\\dk-b63006a2fb9136a7\\out" "-L" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\build\\nrf52840-hal-986d037cc54e058a\\out" "-L" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\build\\nrf52840-pac-94181a918085c019\\out" "-L" "C:\\Users\\mrg\\scoop\\persist\\rustup-gnu\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\thumbv7em-none-eabihf\\lib" "-Bstatic" "C:\\Users\\mrg\\AppData\\Local\\Temp\\rustc6JKiAt\\libcortex_m-88aa8da0118d3940.rlib" "C:\\Users\\mrg\\scoop\\persist\\rustup-gnu\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\thumbv7em-none-eabihf\\lib\\libcompiler_builtins-937035e9954cf71b.rlib" "-Bdynamic" "--eh-frame-hdr" "-z" "noexecstack" "-L" "C:\\Users\\mrg\\scoop\\persist\\rustup-gnu\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\thumbv7em-none-eabihf\\lib" "-o" "C:\\Users\\mrg\\work\\rust-exercises\\nrf52-code\\radio-app\\target\\thumbv7em-none-eabihf\\debug\\deps\\panic-9169f66aa766dd2b" "--gc-sections" "-Tlink.x" "-Tdefmt.x"
  = note:
          flip-link: the native linker failed to link the program normally; please check your project configuration and linker scripts


error: could not compile `radio_app` (bin "panic") due to previous error
```

(Yes, previous exercises have run OK so far.)

Reply:
flip-link is a linker wrapper, and it's hiding the error messages from your actual link. Can you edit the cargo config to not set the linker=flip-link and report the actual linker error?

You are probably missing something else that the lib.rs has that your bin.rs doesn't.

Reply:
I tried this example on Windows and Linux and it was fine.